lib/paperclip/gd_image_processor.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Paperclip
+  class GDImageProcessor < Processor
+    def make
+      original_geometry = Geometry.new(*FastImage.size(file))
+      computed_geometry = original_geometry.resize_to(target_geometry_str)
+      file.rewind
+      original_geometry.to_s[/\d+x\d+/] == computed_geometry.to_s[/\d+x\d+/] ? temp_copy(file) : FastImage.resize(file, computed_geometry.width, computed_geometry.height)
```

Comment:
Question: why the regular expression? is it possible that the geometry is returned in any other format than WxH ?

Reply:
There can be a modifyer character(s) at the end of target geometry.